superset/explore/form_data/commands/update.py

```diff
@@ -75,7 +76,7 @@ def run(self) -> Optional[str]:
                 new_state: TemporaryExploreState = {
                     "owner": owner,
                     "datasource_id": datasource_id,
-                    "datasource_type": datasource_type,
+                    "datasource_type": DatasourceType(datasource_type),
```

Comment:
why is this one different from the above one?

Reply:
No I am referring to this: https://github.com/apache/superset/pull/20149/files/b678a8ba981f7687c17bc68b4f529209bde2d961#diff-6029cd3077bd4123d18311935a0fef83a580ccf22f4a98b6085ed780fd7257e9R27

THis one has `DatasourceType(datasource_type)` the version above has `DatasourceType`